### ISSUE TITLE
property-abstractmethod checking differs from non-property abstractmethod checking

### ISSUE THREAD

**Issue by eamartin** [microsoft/pyright#10268]
**Describe the bug**
Type-checking for property abstractmethods is different than type-checking for non-property abstractmethods.

**Code or Screenshots**

```python
from abc import ABC, abstractmethod
from typing import Generic, TypeVar

T = TypeVar("T")

class Base(ABC, Generic[T]):
    @property
    @abstractmethod
    def prop(self: "Base[int]") -> int: ...

    @abstractmethod
    def meth(self: "Base[int]") -> int: ...


class Impl(Base[T]):
    @property
    def prop(self: "Impl[int]") -> int:
        return 42

    def meth(self: "Impl[int]") -> int:
        return 42
```

Output:
```
pyright_property_bug.py:17:9 - error: "prop" incorrectly overrides property of same name in class "Base"
  __Property method "fget" is incompatible
  ____Parameter 1 type mismatch: base parameter is type "Base[int]", override parameter is type "Impl[int]"
  ______"Base[int]" is not assignable to "Impl[int]" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations
```

I'm not certain but I'd guess that property and method handling should be the same? 

**VS Code extension or command-line**
CLI 1.1.398

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.399.
